src/inline_autocomplete.js

```diff
@@ -0,0 +1,356 @@
+            if (this.completions.filtered.length == 1
+            && this.completions.filtered[0].value == this.prefix
+            && !this.completions.filtered[0].snippet)
+                return this.detach();
+            this.openInlineCompletions();
+            return;
+        }
+        
+        // Save current gatherCompletions session, session is close when a match is inserted
+        var _id = this.gatherCompletionsId;
+
+        var processResults = function(results) {
+            this.prefix = results.prefix;
+            var matches = results.matches;
+
+            this.completions = new FilteredList(matches);
+
+            if (this.exactMatch)
+                this.completions.exactMatch = true;
+
+            this.completions.setPrefix(this.prefix);
+            this.completions.setFilter(this.prefix);
+
+            var filtered = this.completions.filtered;
+
+            if (!results.finished) return this.openInlineCompletions();
+
+
+            if (!filtered.length || (filtered.length == 1 && filtered[0].value === this.prefix)) {
+                return this.detach();
+            }
+
+            this.openInlineCompletions();
+        }.bind(this);
+
+        var isImmediate = true;
+        var immediateResults = null;
+        this.gatherCompletions(function(err, results) {
+            // Wrong prefix or wrong session -> ignore
+            if ((this.prefix && this.prefix.indexOf(results.prefix) !== 0) || _id != this.gatherCompletionsId)
+                return;
+
+            // If multiple completers return their results immediately, we want to process them together
+            if (isImmediate) {
+                immediateResults = results;
+                return;
+            }
+
+            processResults(results);
+        }.bind(this));
+        
+        isImmediate = false;
+        if (immediateResults) {
+            var results = immediateResults;
+            immediateResults = null;
+            processResults(results);
+        }
+    };
+}).call(InlineAutocomplete.prototype);
+
+var destroyCompleter = function(e, editor) {
+    editor.completer && editor.completer.destroy();
+};
+
+InlineAutocomplete.for = function(editor, options) {
+    if (editor.completer) {
+        if (editor.completer instanceof InlineAutocomplete) {
+            return editor.completer;
+        } else {
+            editor.off("destroy", destroyCompleter);
+            editor.completer.destroy();
+        }
+    }
+    editor.completer = new InlineAutocomplete(options);
+    editor.once("destroy", destroyCompleter);
+    return editor.completer;
+};
+
+InlineAutocomplete.startCommand = {
+    name: "startInlineAutocomplete",
+    exec: function(editor, options) {
+        var completer = InlineAutocomplete.for(editor);
+        if (completer instanceof InlineAutocomplete) {
+            completer.showInlineCompletions(editor, options);
+        }
+        return completer;
+    },
+    bindKey: "Alt-C"
```

Comment:
I use mac, and it worked for me with only this, but I agree it is better if it is specified explicitly, rather than relying on an implicit conversion.

Reply:
I meant that with option-c we are very likely to get complaints about some character input on international keyboards not working. At least that's what have happened with our other attempts to use option- on mac.

But since this feature is not enabled by default, i am ok with keeping this as is if that is the desired binding for cloud9